woocommerce/woocommerce-ios
# Use new localization tooling to generate base `.strings` for GlotPress and to download its translations

### Description

Update the Fastlane automation to generated the `.strings` file that gets uploaded to GlotPress and the related logic to download the localized `.strings` to use the new tooling.

It made sense to me to do both sides in one go because of the introduction of the `InfoPlist.strings` files which the `.strings` generation process appends to the base file, and the download process dispatches to the appropriate `.strings` for the app to read a runtime.

Continues the work from #6879. See also [WordPress iOS counterpart](https://github.com/wordpress-mobile/WordPress-iOS/pull/18061).

cc @oguzkocer always appreciate your review, but mostly wanted to flag this to keep an eye out on the `.strings` that will be generated during the next code freeze. I'll also try to check that PR when it's time for it.

### Testing instructions

Manually testing these changes is cumbersome. I don't expect reviewers to do it.

The steps I took to verify these changes were:

1. Create a test branch on top of this, https://github.com/woocommerce/woocommerce-ios/pull/6928, and run both automations on it
2. Create a test branch on top of `trunk` and run both legacy automation steps on it, https://github.com/woocommerce/woocommerce-ios/commit/ac0e8fce93a21181224eb4457856f69a6d2a7750 and https://github.com/woocommerce/woocommerce-ios/commit/188f913a5287f37d8129666bc8dd76e7dcd13a45
3. Ensure that all the changes in the generated `.strings` that the legacy automation produced are present in the new output
4. Compare the `de` `.strings` downloaded by both automations to make sure they are the same. _They weren't_, the new automation correctly removed a key the old one had missed.

---
